examples/bare_barrier.c

```diff
@@ -13,9 +14,9 @@ int main(int argc, char **argv)
     fflush(stdout);
     const char *delay = getenv("DELAY");
     if (delay)
-        setenv(WRITE_VAR, delay, true);
+        assert(setenv(WRITE_VAR, delay, true) == 0);
```

Comment:
Relying on `assert`s here has one drawback: compilation with `-DNDEBUG` will make those statements no-ops. Of course we don't use this flag, but this is why asserts should be rather called on return values, not on the statements themselves. We can afford not checking a return code, but not skipping the whole statement.

Reply:
Sure, I feel a little ashamed now. Using asserts like that is probably first position on every c noob's errors list in existence. Thanks for pointing it out. I will quickly fix it before someone else out there will see it.